test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -126,9 +128,20 @@ class ClusterManagerImplTest : public testing::Test {
     EXPECT_EQ(warming, factory_.stats_.gauge("cluster_manager.warming_clusters").value());
   }
 
+  void checkConfigDump(const std::string& expected_dump_yaml) {
+    auto message_ptr = admin_.config_tracker_.config_tracker_callbacks_["clusters"]();
+    const auto& clusters_config_dump =
+        dynamic_cast<const envoy::admin::v2alpha::ClustersConfigDump&>(*message_ptr);
+
+    envoy::admin::v2alpha::ClustersConfigDump expected_clusters_config_dump;
+    MessageUtil::loadFromYaml(expected_dump_yaml, expected_clusters_config_dump);
+    EXPECT_EQ(expected_clusters_config_dump.DebugString(), clusters_config_dump.DebugString());
```

Comment:
FWIW, we already have in test/test_common/utility.h protoEqual(). I seem to recall there is a way to compare protos with order invariance, but maybe that is only internal.